### DISCUSSION TITLE
contentdefinition.json in Azure Web App (Preview Slots)

### DISCUSSION THREAD

**Discussion by LMSSonos** [OrchardCMS/OrchardCore#6952] [Q&A] [answered]
I have following setup
- website1.azurewebsites.net
- website1-slot1.azurewebsites.net

When making changes to a module or a template. I deploy to the slot1 so I can quickly verify that everything works. The problem is, that the end user on website1.azurewebsites.net has maybe added new Content Types, therefore, `contentdefinition.json` has changed. If I swap the website, the App_Data from slot1 is taken and therefore I have a invalid `contentdefinition.json` file on the website1. 

Currently I manually copy contentdefinition.json from website1 to website1-slot1 before I make the swap. Is there any better option to have this all stored into the database or to force recreation?

Thanks

**Comment by ns8482e** [comment, marked as answer]
@LMSSonos Store content definition in SQL. See this guide https://docs.orchardcore.net/en/latest/docs/guides/content-definitions/#understanding-content-definition-stores

**Reply by LMSSonos** [reply]
Thanks, exactly what I was looking for
